tests/test_util.py

```diff
@@ -456,3 +460,49 @@ def test_df_parser(tmpdir):
     assert len(df.labels) == 1
     assert df.labels.get('label') == 'foobar barfoo'
 
+def test_df_parser_parent_env_arg(tmpdir):
+    tmpdir_path = str(tmpdir.realpath())
+    p_env = {
+        "test_env": "first"
+    }
+    df_content = dedent("""\
+        FROM fedora
+        ENV foo=bar
+        LABEL label="foobar $test_env"
+        """)
+
+    with open(tmpdir_path + '/Dockerfile', 'w') as dockerfile:
+        dockerfile.write(df_content)
+        dockerfile.flush()
+
+    df = df_parser(tmpdir_path, parent_env=p_env)
+    assert df.labels.get('label') == 'foobar first'
+
+@pytest.mark.parametrize('env_arg', [
+    {"test_env": "first"},
+    ['test_env=first'],
```

Comment:
Why would env vars lib contain lists?

Also, in `pre_distribution_scope` `base_image_inspect` contains a dict 'Labels' - is this test correct?